core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -122,11 +116,14 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
     override def fromConfigObject(co: ConfigObject, default: Option[V] :: U): Try[FieldType[K, V] :: T] = {
       val keyStr = mapping(key.value.toString().tail)
       for {
-        v <- improveFailure(
-          vFieldConvert.value.from(co.get(keyStr)) match {
-            case Failure(CannotConvertNullException) =>
+        v <- improveFailure[V](
+          (co.get(keyStr), vFieldConvert.value) match {
+            case (null, converter: OptionConfigConvert[_]) =>
```

Comment:
I like that idea, and I've implemented it in 971c2c7. I named the trait `AllowMissingKey`, though. I don't think it is necessary to put `ConfigConvert` in the name, as it will already be mixed into one, and I think `AllowMissingKey` is clearer than `AllowMissing` or `AllowNull`, but I'm open to change it.